Chapter6/link_ping_pong.erl

```diff
@@ -0,0 +1,60 @@
+-module(link_ping_pong).
+-export([start/0, print/1, stop/0]).
+-export([init/0]).
+
+% Exercise 6-1: The Linked Ping Pong Server
+
+start() -> register(msg_server, spawn(link_ping_pong, init, [])).
+
+print(Message) ->
+	msg_server!{message, Message, self()},
+	process_flag(trap_exit, true),
+	link(whereis(msg_server)),
+	receive
+		{'EXIT', _Pid, Reason} -> {error, Reason};
+		{reply, Reply} -> Reply
+	end.
+
+% Link method
+stop() ->
+	msg_server!{stop, self()},
+	process_flag(trap_exit, true),
+	link(whereis(msg_server)),
```

Comment:
I guess caller process already linked itself to `msg_server` when it called `print/1`. You don't need to link it again here. Moreover, if it was the only place where it's linked to `msg_server`, then you'd need to link it first and then send `stop` message since at the time you call link to the process ID it might be already dead (unless you did this on purpose).